File: lihua-biz/lihua-system/src/main/java/com/lihua/entity/SysNotice.java
package com.lihua.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.lihua.mybatis.model.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = true)
@Data
public class SysNotice extends BaseEntity {

    /**
     * 主键 id
     */
    @TableId(type = IdType.ASSIGN_ID)
    private String id;

    /**
     * 标题
     */
    private String title;

    /**
     * 类型
     */
    private String type;

    /**
     * 状态： 未发布 0 / 已发布 1 / 已撤销 2
     */
    private String status;

    /**
     * 优先级
     */
    private String priority;

    /**
     * 是否发送给全部用户
     */
    private String userScope;

    /**
     * 内容
     */
    private String content;

    /**
     * 发布时间
     */
    private LocalDateTime releaseTime;

    /**
     * 发布人id
     */
    private String releaseId;

    /**
     * 备注
     */
    private String remark;

    /**
     * 图标
     */
    private String icon;

}

File: lihua-vue/src/api/system/notice/type/sys-notice.ts
export interface SysNotice {
    /**
     * 主键 id
     */
    id?: string;

    /**
     * 标题
     */
    title?: string;

    /**
     * 类型
     */
    type?: string;

    /**
     * 状态： 未发布 0 / 已发布 1 / 已撤销 2
     */
    status?: string;

    /**
     * 优先级
     */
    priority?: string;

    /**
     * 发送的用户范围
     */
    userScope?: string;

    /**
     * 内容
     */
    content?: string;

    /**
     * 发布时间
     */
    releaseTime?: Date;

    /**
     * 逻辑删除标识
     */
    delFlag?: string;

    /**
     * 备注
     */
    remark?: string;

    /**
     * 创建时间
     */
    createTime?: Date;

    /**
     * 创建人id
     */
    createId?:string;

    /**
     * 图标
     */
    icon?: string;
}

export interface SysNoticeVO extends SysNotice {
    /**
     * 指定用户
     */
    userIdList?: string[];

    /**
     * 创建用户
     */
    createUser?: string;

    /**
     * 发布用户
     */
    releaseUser?: string;
}

export interface SysNoticeDTO extends SysNotice {
    /**
     * 指定用户
     */
    userIdList?: string[];

    /**
     * 当前页数
     */
    pageNum: number;

    /**
     * 每页记录数
     */
    pageSize: number;

    /**
     * 是否查询star数据
     */
    star?: '1'
}
